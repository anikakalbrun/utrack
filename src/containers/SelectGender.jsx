import React, {Component} from 'react';
import {string, func} from 'prop-types';

import FilterWrapper from '../components/FilterWrapper';
import Button from '../components/Button';

const getBtnClass = (value, btnText) => value === btnText ? "btn-success" : "btn-light";
const isBtnDisabled = (btnClass) => btnClass === "btn-success";

class SelectGender extends Component {

    static propTypes = {
        value: string,
        setGenderFilter: func
    };

    static defaultProps = {
        value: 'All'
    };

    handleClickMaleBtn = () => {
        this.props.setGenderFilter('Male');
    };

    handleClickFemaleBtn = () => {
        this.props.setGenderFilter('Female');
    };

    handleClickAllBtn = () => {
        this.props.setGenderFilter('All');
    };

    render() {
        const {value} = this.props;
        return (
            <FilterWrapper label='Gender'>
                <div className='empty'/>
                <div className="btn-group" role="group" aria-label="gender">
                    <Button text='Male'
                            btnClass={getBtnClass(value, 'Male')}
                            disabled={isBtnDisabled(getBtnClass(value, 'Male'))}
                            handleClick={this.handleClickMaleBtn}/>
                    <Button text='Female'
                            btnClass={getBtnClass(value, 'Female')}
                            disabled={isBtnDisabled(getBtnClass(value, 'Female'))}
                            handleClick={this.handleClickFemaleBtn}/>
                    <Button text='All'
                            btnClass={getBtnClass(value, 'All')}
                            disabled={isBtnDisabled(getBtnClass(value, 'All'))}
                            handleClick={this.handleClickAllBtn}
                    />
                </div>
            </FilterWrapper>
        )
    }
}

export default SelectGender