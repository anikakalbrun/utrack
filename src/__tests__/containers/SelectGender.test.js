import React from 'react';
import SelectGender from '../../containers/SelectGender';

let wrapper;
const setGenderFilter = jest.fn();
const props = {
    setGenderFilter,
    value: 'All'
};

describe('RangeInput', () => {

    beforeEach(() => {
        wrapper = shallow(<SelectGender {...props}/>);
    });

    it('renders FilterWrapper', () => {
        expect(wrapper.find('FilterWrapper').length).toEqual(1);
    });

    it('passes the label property to FilterWrapper', () => {
        expect(wrapper.find('FilterWrapper').props().label).toEqual('Gender');
    });

    it('renders a div with class empty', () => {
        expect(wrapper.find('div.empty').length).toEqual(1);
    });

    it('renders a div with class btn-group, role="group", aria-label="gender"', () => {
        expect(wrapper.find('div.btn-group[role="group"][aria-label="gender"]').length).toEqual(1);
    });

    it('renders a three buttons', () => {
        expect(wrapper.find('Button').length).toEqual(3);
    });

    it('passes proper props to the first button', () => {
        expect(wrapper.find('Button').at(0).props()).toMatchSnapshot();
    });

    it('passes proper props to the second button', () => {
        expect(wrapper.find('Button').at(1).props()).toMatchSnapshot();
    });

    it('passes proper props to the third button', () => {
        expect(wrapper.find('Button').at(2).props()).toMatchSnapshot();
    });

    it('clicking the first button calls setGenderFilter withe the value "Male"', () => {
        wrapper.find('Button').at(0).dive().find('button').simulate('click');
        expect(setGenderFilter).toHaveBeenCalledWith('Male');
    });

    it('clicking the second button calls setGenderFilter withe the value "Female"', () => {
        wrapper.find('Button').at(1).dive().find('button').simulate('click');
        expect(setGenderFilter).toHaveBeenCalledWith('Female');
    });

    it('clicking the third button calls setGenderFilter withe the value "All"', () => {
        wrapper = shallow(<SelectGender setGenderFilter={setGenderFilter}
                                        value={'Male'}/>);
        wrapper.find('Button').at(2).dive().find('button').simulate('click');
        expect(setGenderFilter).toHaveBeenCalledWith('All');
    });
});
