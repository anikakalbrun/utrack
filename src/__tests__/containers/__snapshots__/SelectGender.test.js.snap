// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RangeInput passes proper props to the first button 1`] = `
Object {
  "btnClass": "btn-light",
  "disabled": false,
  "handleClick": [Function],
  "text": "Male",
  "type": "button",
}
`;

exports[`RangeInput passes proper props to the second button 1`] = `
Object {
  "btnClass": "btn-light",
  "disabled": false,
  "handleClick": [Function],
  "text": "Female",
  "type": "button",
}
`;

exports[`RangeInput passes proper props to the third button 1`] = `
Object {
  "btnClass": "btn-success",
  "disabled": true,
  "handleClick": [Function],
  "text": "All",
  "type": "button",
}
`;
